### ISSUE TITLE
`dev --local Foo` not respected if `Foo` exists

### ISSUE THREAD

**Issue by ssfrr** [JuliaLang/Pkg.jl#783]
MWE:

```
(v1.0) pkg> dev Example
   Cloning git-repo `https://github.com/JuliaLang/Example.jl.git`
  Updating git-repo `https://github.com/JuliaLang/Example.jl.git`
 Resolving package versions...
  Updating `C:\Users\sfr\.julia\environments\v1.0\Project.toml`
  [7876af07] + Example v0.5.1+ [`C:\Users\sfr\.julia\dev\Example`]
  Updating `C:\Users\sfr\.julia\environments\v1.0\Manifest.toml`
  [7876af07] + Example v0.5.1+ [`C:\Users\sfr\.julia\dev\Example`]

(v1.0) pkg> activate --shared ExampleDev
[ Info: new shared environment "ExampleDev" will be placed at C:\Users\sfr\.julia\environments\Examp
leDev

(ExampleDev) pkg> dev --local Example
 Resolving package versions...
  Updating `C:\Users\sfr\.julia\environments\ExampleDev\Project.toml`
  [7876af07] + Example v0.5.1+ [`C:\Users\sfr\.julia\environments\ExampleDev\..\..\dev\Example`]
  Updating `C:\Users\sfr\.julia\environments\ExampleDev\Manifest.toml`
  [7876af07] + Example v0.5.1+ [`C:\Users\sfr\.julia\environments\ExampleDev\..\..\dev\Example`]
  [2a0f44e3] + Base64
  [8ba89e20] + Distributed
...
```

I expected `Example` to be inside `environments\ExampleDev\dev` but it used the one in my global `dev` directory instead. It does the same thing even if you `(v1.0) pkg> rm Example`.

**Comment by fredrikekre** [comment]
I can not reproduce this, I get `Example` in `ExampleDev/dev/Example`. Maybe you used Julia v1.0.0? (versioninfo() is always useful...). Possibly fixed by https://github.com/JuliaLang/Pkg.jl/pull/657.

**Comment by ssfrr** [comment]
sorry for forgetting the versioninfo. Yep, seems to be fixed in 1.0.1. thanks!
